### PR TITLE
docs: translate file src/guide/essentials/computed.md

### PR DESCRIPTION
**Pull request by nguyenvanduocit**

### CODE REVIEW

**Comment by nguyenvanduocit** [comment]
https://deploy-preview-17--vue-vi.netlify.app/guide/essentials/computed.html

File: src/guide/essentials/computed.md [hunk 1]
```diff
@@ -120,19 +120,19 @@ const publishedBooksMessage = computed(() => {
...
 
-Here we have declared a computed property `publishedBooksMessage`. The `computed()` function expects to be passed a getter function, and the returned value is a **computed ref**. Similar to normal refs, you can access the computed result as `publishedBooksMessage.value`. Computed refs are also auto-unwrapped in templates so you can reference them without `.value` in template expressions.
+Ở đây chúng ta đã khai báo một thuộc tính computed `publishedBooksMessage`. Hàm `computed()` nhận vào một hàm getter (hàm dùng để lấy giá trị), và giá trị được trả về là một **computed ref (tham chiếu)**. Tương tự như các ref thông thường, bạn có thể truy cập giá trị của nó thông qua `publishedBooksMessage.value`. Khi ref được dùng trong template, nó sẽ được unwrap tự động nên bạn có thể tham chiếu tới nó mà không cần `.value`.
```
Comment (phanan) [inline src/guide/essentials/computed.md:125]:
```suggestion
Ở đây chúng ta đã khai báo một thuộc tính computed `publishedBooksMessage`. Hàm `computed()` nhận vào một hàm getter (hàm dùng để lấy giá trị), và giá trị được trả về là một **computed ref (tham chiếu)**. Tương tự như các ref thông thường, bạn có thể truy cập giá trị của nó thông qua `publishedBooksMessage.value`. Khi dùng trong template, `ref`sẽ được unwrap tự động nên bạn có thể tham chiếu tới nó mà không cần `.value`.
```

File: src/guide/essentials/computed.md [hunk 2]
```diff
@@ -120,19 +120,19 @@ const publishedBooksMessage = computed(() => {
...
 
-A computed property automatically tracks its reactive dependencies. Vue is aware that the computation of `publishedBooksMessage` depends on `author.books`, so it will update any bindings that depend on `publishedBooksMessage` when `author.books` changes.
+Một thuộc tính computed sẽ tự động theo dõi các sự phụ thuộc có tính reactive. Vue ý thức rằng sự tính toán của `publishedBooksMessage` phụ thuộc vào `author.books`, nên nó sẽ cập nhật mọi ràng buộc vào `publishedBooksMessage` khi `author.books` thay đổi.
```
Comment (phanan) [inline src/guide/essentials/computed.md:127]:
```suggestion
Một thuộc tính computed sẽ tự động theo dõi các phụ thuộc có tính reactive. Vue biết rằng giá trị của `publishedBooksMessage` phụ thuộc vào `author.books`, nên nó sẽ cập nhật mọi ràng buộc vào `publishedBooksMessage` khi `author.books` thay đổi.
```

File: src/guide/essentials/computed.md [hunk 3]
```diff
@@ -186,13 +186,13 @@ const now = computed(() => Date.now())
 
 </div>
 
-In comparison, a method invocation will **always** run the function whenever a re-render happens.
+Trong khi đó, một lệnh gọi phương thức sẽ **luôn** chạy hàm đó bất cứ khi nào việc render lại xảy ra.
 
-Why do we need caching? Imagine we have an expensive computed property `list`, which requires looping through a huge array and doing a lot of computations. Then we may have other computed properties that in turn depend on `list`. Without caching, we would be executing `list`’s getter many more times than necessary! In cases where you do not want caching, use a method call instead.
+Tại sao chúng ta cần cache? Tưởng tượng rằng chúng ta có một thuộc tính computed nặng nề là `list`, nó đòi hỏi phải lặp qua một array lớn và thực hiện rất nhiêu phép tính. Sau đó, chúng ta có thể có những thuộc tính computed khác phụ thuộc vào `list`. Nếu không có cache, chúng ta sẽ phải thực thi hàm getter của `list` nhiều lần hơn mức cần thiết! Trong những trường hợp bạn không muốn cache, hãy dùng lệnh gọi phương thức.
```
Comment (phanan) [inline src/guide/essentials/computed.md:191]:
```suggestion
Tại sao chúng ta cần cache? Giả sử ta có một thuộc tính computed là `list`, thuộc tính này đòi hỏi lặp qua một mảng khổng lồ và thực hiện rất nhiều phép tính. Tiếp theo đó, chúng ta có thể có những thuộc tính computed khác phụ thuộc vào `list`. Nếu không có cache, chúng ta sẽ phải thực thi hàm getter của `list` nhiều hơn mức cần thiết rất nhiều! Trong những trường hợp bạn không muốn cache, hãy sử dụng phương thức thay vì thuộc tính computed.
```

File: src/guide/essentials/computed.md [hunk 4]
```diff
@@ -186,13 +186,13 @@ const now = computed(() => Date.now())
...
 
-## Writable Computed
+## Thuộc tính computed có thể ghi
 
-Computed properties are by default getter-only. If you attempt to assign a new value to a computed property, you will receive a runtime warning. In the rare cases where you need a "writable" computed property, you can create one by providing both a getter and a setter:
+Mặc định thì những thuộc tính computed chỉ có getter. Nếu bạn cố ghi giá trị mới vào một thuộc tính computed, bạn sẽ nhận được cảnh báo. Trong các trường hợp hiếm hoi mà bạn cần một thuộc tính computed có thể ghi, bạn có thể cung cấp cả getter và setter cho nó:
```
Comment (phanan) [inline src/guide/essentials/computed.md:195]:
```suggestion
Mặc định thì những thuộc tính computed là chỉ đọc (chỉ có getter). Nếu bạn cố ghi giá trị mới vào một thuộc tính computed, bạn sẽ nhận được một cảnh báo runtime. Trong các trường hợp hãn hữu bạn cần một thuộc tính computed có thể ghi, bạn có thể cung cấp cả getter và setter cho nó:
```

File: src/guide/essentials/computed.md [hunk 5]
```diff
@@ -247,16 +247,16 @@ const fullName = computed({
 </script>
 ```
 
-Now when you run `fullName.value = 'John Doe'`, the setter will be invoked and `firstName` and `lastName` will be updated accordingly.
+Giờ khi bạn chạy `fullName.value = 'John Doe'`, setter sẽ được gọi rồi `firstName` vả `lastName` sẽ được cập nhật tương ứng.
 
 </div>
 
 ## Best Practices
```
Comment (phanan) [inline src/guide/essentials/computed.md:254]:
```suggestion
## Lời khuyên khi dùng thuộc tính computed 
```

File: src/guide/essentials/computed.md [hunk 6]
```diff
@@ -247,16 +247,16 @@ const fullName = computed({
...
 
-### Getters should be side-effect free
+### Getter không nên có hiệu ứng phụ
 
-It is important to remember that computed getter functions should only perform pure computation and be free of side effects. For example, don't make async requests or mutate the DOM inside a computed getter! Think of a computed property as declaratively describing how to derive a value based on other values - its only responsibility should be computing and returning that value. Later in the guide we will discuss how we can perform side effects in reaction to state changes with [watchers](./watchers).
+Điều quan trọng cần phải nhớ là hàm getter chỉ nên thực hiện duy nhất một việc tính toán đơn thuần và không có hiệu ứng phụ. Ví dụ, đừng tạo một request đồng bộ hoặc thay đổi DOM bên trong một hàm getter! Hãy nghĩ rằng thuộc tính computed như là một khai báo về cách để lấy giá trị dựa trên những giá trị khác - nó chỉ có trách nhiệm tính toán và trả về giá trị đó mà thôi. Ở phần sau của hướng dẫn, chúng ta sẽ thảo luận về cách để thực hiện một hiệu ứng phụ dựa trên sự thay đổi của state bằng [watchers](./watchers).
```
Comment (phanan) [inline src/guide/essentials/computed.md:258]:
```suggestion
Điều quan trọng cần phải nhớ là hàm getter chỉ nên thực hiện duy nhất một việc tính toán đơn thuần và không có hiệu ứng phụ. Ví dụ, đừng tạo một request đồng bộ hoặc thay đổi DOM bên trong một hàm getter! Hãy xem thuộc tính computed như một khai báo về cách để lấy giá trị dựa trên những giá trị khác – nó chỉ có trách nhiệm tính toán và trả về giá trị đó mà thôi. Ở phần sau của hướng dẫn, chúng ta sẽ thảo luận về cách để thực hiện một hiệu ứng phụ dựa trên sự thay đổi của state bằng [watcher](./watchers). 
```

**Review by phanan — COMMENTED** [review]
👍🏼

**Pull request merged**